odb_extractor_atoms.py

```diff
@@ -0,0 +1,627 @@
+        step_time = step['totalTime']/1000.
+        
+        if 'GMOS' in inst:
+            coadds = 1
+        else:
+            coadds = int(step['observe:coadds'])
+
+        disperser = step['instrument:disperser']
+
+        # Offsets
+        if 'telescope:p' in keys:
+            p = float(step['telescope:p'])
+        else:
+            p = 0.0
+        if 'telescope:q' in keys:
+            q = float(step['telescope:q'])
+        else:
+            q = 0.0
+        qoffsets.append(q)
+
+        # Guiding
+        guiding = 'park'
+        for key in keys:
+            if 'guideWith' in key:
+                if step[key] != 'park':
+                    guiding = step[key]
+                    break
+
+        # Any wavelength/filter_name change is a new atom
+        if ii == 0 or (ii > 0 and wavelength != float(sequence[ii-1]['instrument:observingWavelength'])):
+            nextatom = True
+            print('Atom for wavelength')
+                
+#         if observe_class == 'science':
+            
+        # AB
+        # ABBA
+        if nsteps >= 4 and nsteps - ii > 3 and nabba == 0:
+            # print(q, sequence[ii+1]['telescope:q'], sequence[ii+2]['telescope:q'], sequence[ii+3]['telescope:q'])
+            if q == float(sequence[ii+3]['telescope:q']) and q != float(sequence[ii+1]['telescope:q']) and                float(sequence[ii+1]['telescope:q']) == float(sequence[ii+2]['telescope:q']):
+                    nabba = 3
+                    nextatom = True
+                    print('Atom for ABBA')
+        else:
+            nabba -= 1
+        
+        if nextatom:
+            natom += 1
+            atoms.append({'id': natom, 'exec_time': 0.0, 'prog_time': 0.0, 'part_time': 0.0, 'filter': filter_name,
+                          'fpu': fpu, 'disperser': disperser, 'wavelength': wavelength, 'guiding': guiding})
+
+        atoms[-1]['exec_time'] += step_time
+        
+        atomlabel = natom
+        if 'partnerCal' in observe_class:        
+            atomlabel *= 10
+            atoms[-1]['part_time'] += step_time
+        else:
+            atoms[-1]['prog_time'] += step_time
+            
+#         atoms[-1]['id'] = atomlabel
+            
+        print('{:22} {:12} {:7.2f} {:3d} {:10} {:15} {:12} {:12} {:7.4f} {:5.2f} {:5.2f} {:3d}'.format(datalab,
+               observe_class, exptime, coadds, inst, fpu, filter_name, disperser, wavelength, p, q, atomlabel))
+
+    return atoms
+
+
+def printseq(sequence, comment='', csv=False, path=''):
+    '''Print basic configuration information about a sequence, with an option to write to a csv file'''
+
+    atom = '1'
+    if csv and path != '':
+        obsid = sequence[0]['ocs:observationId']
+        filename = os.path.join(path, obsid + '_seq.csv')
+        f = open(filename, 'w')
+        print('{},{}'.format('comment', comment), file=f)
+        print('{},{},{},{},{},{},{},{},{},{},{},{}'.format('datalab', 'class', 'exptime', 'coadds', 'inst', 'fpu',
```

Comment:
In Python 3, this could be written:
```
print(f'{datalab}{class}{exptime}{coadds}{inst}{fpu}{filter_name}{disperser}{wavelength}{p}{q}{atom}', file=f)
```